### PR TITLE
unwrap valHash to raw data in scanCollection

### PR DESCRIPTION
**Pull request by brucexhan**

### CODE REVIEW

File: apps/nosql/nosql.html [hunk 1]
```diff
@@ -830,31 +830,20 @@ <h5 class="modal-title">Create Account <span id="createAccountMsg"></span></h5>
 
 					for (let i=(currPageNumber - 1)*itemCountPerPage; i<currPageNumber*itemCountPerPage; i++) {
 						if (!dataRows[i]) continue;
-						keyValhashObj[dataRows[i].key] = {"size": dataRows[i].size, "valHash": dataRows[i].valHash};
-						keyArr.push(dataRows[i].key);
+						keyValhashObj[dataRows[i].Nonce] = {"Nonce": dataRows[i].Nonce, "key": dataRows[i].key, "size": dataRows[i].size, "valHash": dataRows[i].valHash};
+						keyArr.push(dataRows[i].Nonce);
```
Comment (brucexhan) [inline apps/nosql/nosql.html]: sort the scanCollection result by nonce

File: apps/nosql/nosql.html [hunk 2]
```diff
@@ -830,31 +830,20 @@ <h5 class="modal-title">Create Account <span id="createAccountMsg"></span></h5>
...
-							}
-	
-						});						
+						dataRows[j] = keyValhashObj[keyArr[j]];
+						dataRows[j].value =  wolkjs.getValue(keyValhashObj[keyArr[j]].valHash);
+						
+						rows += scanCollNewTableRow(dataRows[j]);
 					}
 
+					document.getElementById("scanCollNewDiv").innerHTML = scanCollNewTable(rows);
```
Comment (brucexhan) [inline apps/nosql/nosql.html]: using getKey (which is an asynchronous function) to get raw value of a key is not compatible in the auto-refreshing mode. it brings 2 issues as I notice:
a. the result sorted by key/nonce cannot be maintained due to the delay from asynchronousness
b. event listeners cannot be attached before next round of refreshing due to the delay from asynchronousness

so I basically revert this part to old approach [https://github.com/wolkdb/wolkjs/blob/6b420703353561f5f2983aa988daca3af1699574/apps/nosql/nosql.html#L1041-L1056] of using valHash/chunkID to fetch the raw data, but this time I extract it to a getValue function in wolk.js (as below)

File: apps/wolk.js [hunk 3]
```diff
@@ -648,6 +648,12 @@ class WOLK {
   getTransaction(txhash, callback) {
     return this.get(joinPath('wolk', 'tx', txhash), {}, callback);
   }
+  getValue(valHash) {
+    var request = new XMLHttpRequest();
+    request.open('GET', `${this.provider}/wolk/chunk/${valHash}`, false);
+    request.send(null);
+    return request.responseText;
+  }
```
Comment (brucexhan) [inline apps/wolk.js]: mentioned above. extracted from nosql.html

Comment (rodneywitcher) [reply apps/wolk.js]: Looks like there is a conflict in the code.  Can you merge `dev` into `nosql-bh` (if necessary) and also fix the conflict then push nosql-bh.

In apps/wolk.js

```
<<<<<<< nosql-bh
  getValue(valHash) {
    var request = new XMLHttpRequest();
    request.open('GET', `${this.provider}/wolk/chunk/${valHash}`, false);
    request.send(null);
    return request.responseText;
  }
  transfer(recipient, amount, callback) {
=======
  transfer(recipient, amount, options, callback) {
>>>>>>> dev
```

Once you merge dev into nosql-bh, you will likely be able to resolve and then re-push.

Comment (brucexhan) [reply apps/wolk.js]: ok. I want to merge dev with PR#86 into nosql-bh. then resolve from there

Comment (rodneywitcher) [reply apps/wolk.js]: ok ... sounds good.  Will merge #86 as soon as you give the go ahead.

**Pull request merged**